catalog-scripts/idp-catalog-wizard.py

```diff
@@ -0,0 +1,174 @@
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  tags:
+    - java
+    - harness
+    - auto-generated
+  name: {repo_name}
+  annotations:
+    backstage.io/source-location: url:{repo_path}
+spec:
+  type: service
+  system: harness
+  lifecycle: experimental
+  owner: Harness_Account_All_Users
+"""
+
+def get_repositories_api(organization, token):
+    url = f"https://api.github.com/orgs/{organization}/repos"
+    headers = {
+        "Accept": "application/vnd.github.v3+json",
+        "Authorization": f"Bearer {token}",
+        "X-GitHub-Api-Version": "2022-11-28"
+    }
+    response = requests.get(url, headers=headers)
+    return response
+
+def list_repositories(organization, token, repo_pattern=None):
+    response = get_repositories_api(organization, token)
+
+    yaml_files_created = 0
+
+    if response.status_code == 200:
+        repos = response.json()
+        print(f"Repositories in {organization}:")
+        for repo in repos:
+            repo_name = repo['name']
+            if repo_name == current_directory:
+                continue
+            repo_path = repo['html_url']
+            if repo_pattern is None or re.match(repo_pattern, repo_name):
+                print(repo_name)
+                create_or_update_catalog_info(repo_name, repo_path)
+                yaml_files_created += 1
+        print("----------")
+    else:
+        print(f"Failed to fetch repositories. Status code: {response.status_code}")
+
+    print(f"Total YAML files created or updated: {yaml_files_created}")
+
+def create_or_update_catalog_info(repo_name, repo_path):
+    directory = f"services/{repo_name}"
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+    
+    yaml_file_path = f"{directory}/catalog-info.yaml"
+
+    if os.path.exists(yaml_file_path):
+        # Update existing YAML file
+        with open(yaml_file_path, "r") as file:
+            existing_content = file.read()
+        updated_content = yaml_content_template.format(repo_name=repo_name, repo_path=repo_path)
+        with open(yaml_file_path, "w") as file:
+            file.write(updated_content)
+    else:
+        # Create new YAML file
+        with open(yaml_file_path, "w") as file:
+            file.write(yaml_content_template.format(repo_name=repo_name, repo_path=repo_path))
+
+branch  = "main"
+def register_yamls(organization, account, x_api_key):
+    # Placeholder function for registering YAML files
+    print("Registering YAML files...")
+    count = 0
+    api_url = f"https://idp.harness.io/{account}/idp/api/catalog/locations"
+
+
+    repos = [name for name in os.listdir("./services") if os.path.isdir(os.path.join("./services", name))]
+    for repo_name in repos:
+        if repo_name != current_directory:
+            directory = f"services/{repo_name}"
+            api_payload = {
+                "target": f"https://github.com/{organization}/{current_directory}/blob/{branch}/{directory}/catalog-info.yaml",
```

Comment:
Are we guessing the repo name based on the current_directory ? 🤔 should this be an option as well

Reply:
The repo is to be cloned and then we will be running the scripts at the root. Hence, current dir was used. Should we use the repo name instead?

Reply:
that's okay then. we need the instructions and an example to be honest